src/components/c-date-picker.vue

```diff
@@ -215,16 +214,17 @@
           return;
         }
 
-        if (this.$dayjs(value).format() !== this.$dayjs(oldValue).format()) {
-          if (this.range) {
-            this.calendarInstance.setStartRange(value);
-            this.calendarInstance.draw();
-          } else {
-            this.close();
-          }
+        if (this.$dayjs(value).format() === this.$dayjs(oldValue).format()) {
+          return;
+        }
 
-          this.onChange();
+        if (this.range) {
+          this.calendarInstance.setStartRange(value);
+          this.calendarInstance.draw();
+        } else {
+          this.close();
         }
```

Comment:
todo: please always add an empty line before and after a `block` (section with curved brackets `{}`).